scripts/common/shell_command.py

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env python3
+import os
+import subprocess
+
+def run_cmd(command: str, env: dict = None, logfile: str = None) -> int:
```

Comment:
I don't think we really need this. I mean, it would add some value if it was customized for some specific use case but here it seems to be just boilerplate on top of the built-in `subprocess.run()` that unnecessarily hides some details.

For example it runs all commands in the shell while it's generally safer to only use shell when really needed. With the way you invoke the it (`run_cmd(f"{test_script_path} {solc_binary_type} {solc_binary_path}")`), it would be trivial to avoid shell here.

Reply:
hum..fair enough. It is because I use this a lot in the Foundry PR as well, see: https://github.com/ethereum/solidity/pull/13873/files#diff-b42a8d824f3a03275495b8105ea11eb66d6cd562d5088c588319520ca0dd2132R144

But yeah, maybe it is better to directly use subprocess there as well.

Reply:
Well, my main beef with such a wrapper is that it's basically going back to the time of `os.system()`, which is [discouraged nowadays for very good reasons](https://stackoverflow.com/questions/51329108/is-using-os-systemcurl-in-python-truly-unsafe-compared-to-native-librar).

Also, many of the arguments are actually commonly used and the wrapper will inevitably evolve into something with tons of parameters like the original function or grow multiple variants to cover all use cases.

If you really want to make it less verbose in the Foundry PR, my suggestion would be to instead prepare small specialized wrapper for its commands, just like I did with git in [`git_helpers.py`](https://github.com/ethereum/solidity/blob/develop/scripts/common/git_helpers.py). This way you can hide any extra arguments, include some output processing if necessary, run sanity checks that it did what you expected it to, etc.

Alternatively, specifically in #13873 where the commands are actually very simple, you could keep the wrapper local to the script so that you can make local assumptions and actually hide unused parameters. But I'd still avoid running the command through shell.

Reply:
Got it. I removed the wrapper and I will update the other PRs that make use of it ;)